src/rust/engine/src/scheduler.rs

```diff
@@ -310,11 +315,28 @@ impl Scheduler {
       .collect()
   }
 
-  fn display_ongoing_tasks(graph: &Graph<NodeKey>, roots: &[NodeKey], display: &mut EngineDisplay) {
+  fn display_ongoing_tasks(
+    graph: &Graph<NodeKey>,
+    roots: &[NodeKey],
+    display: &mut EngineDisplay,
+    tasks_to_display: &mut IndexMap<String, Duration>,
+  ) {
+    // Update the graph. To do that, we iterate over heavy hitters.
+    let heavy_hitters: Vec<(String, Duration)> =
+      graph.heavy_hitters(&roots, display.worker_count());
+    // Insert every one in the set of tasks to display.
+    tasks_to_display.extend(heavy_hitters.iter().cloned());
+    // And remove the tasks that no longer should be there.
+    for (task, duration) in tasks_to_display.clone().into_iter() {
+      if !heavy_hitters.contains(&(task.to_string(), duration)) {
```

Comment:
On every run, the duration of a task will change... so I'm not sure that this is doing the right thing. It's possible that it should only be comparing the task name?

This might also be n^2 (small N, of course, but) because `Vec::contains` will be linear.

Reply:
Making `heavy_hitters` into a `HashMap` solved both problems (constant lookup, lookup by key). Since we don't actually care about the order of `heavy_hitters` anymore, it should be fine.